Sources/PitchSpeller/Data Structures/FlowNetwork.swift

```diff
@@ -32,18 +39,78 @@ public struct FlowNetwork <Value: Hashable>: Hashable {
     /// reached zero in the flow-propagation process.
     ///
     /// - TODO: Add backflow to reversed edges.
-    public var residualNetwork: Graph<Value> {
+    public var residualNetwork: Graph<Node> {
         var residualNetwork = graph
         while let path = residualNetwork.shortestPath(from: source, to: sink) {
             residualNetwork.insertPath(path.map { $0 - maximumFlow(of: path) })
         }
         return residualNetwork
     }
+    
+    /// - Returns: (0) The maximum flow of the network and (1) the residual network produced after
+    /// pushing all possible flow from source to sink (while satisfying flow constraints) - with
+    /// saturated edges flipped and all weights removed.
+    var solvedForMaximumFlow: (Double, UnweightedGraph<Node>) {
+        var totalFlow = 0.0
+        var residualNetwork = directedGraph
+        
+        func findAugmentingPath () -> Bool {
+            guard let path = residualNetwork.shortestUnweightedPath(from: source, to: sink) else {
+                return false
+            }
+            pushFlow(through: path)
+            return true
+        }
+            
+        func pushFlow (through path: UnweightedGraph<Node>.Path) {
+            let minimumEdge = (path.adjacents.compactMap {
+                residualNetwork.weight($0)
+                }.min())!
+            totalFlow += minimumEdge
+            path.adjacents.forEach {
+                residualNetwork.updateEdge($0, with: {
```

Comment:
```
residualNetwork.updateEdge($0, with: { $0 - minimumEdge })
```

Reply:
Yeah the reason I did the more verbose closure was to avoid ambiguity with two `$0` placeholders meaning different things. First `$0` is a `Pair` and second is a `Weight`

Reply:
Ahh, that's fair. Maybe then I'd go full verbose:

```Swift
path.adjacents.forEach { edge in 
    residualNetwork.updateEdge(edge, with: { value in value - minimumEdge })
}
```